File: MinimumSizeSubarraySum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int total = 0;
        int res = nums.length + 1;
        for (int i = 0; i <  nums.length; i++){
            total += nums[i];
            while (total >= target){
                res =  Math.min(i - left + 1, res);
                total -= nums[left];
                left += 1;
                
            }
        }
        if (res == nums.length + 1) return 0;
        else return res;
    }
}

File: MinimumSizeSubarraySum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        #sliding window
        min_sum_length = len(nums) + 1
        left = 0
        value = 0
        for right in range(len(nums)):
            value += nums[right]

            while value >= target:
                min_sum_length = min(min_sum_length, right - left + 1)
                value -= nums[left]
                left += 1
        
        if min_sum_length == len(nums) + 1:
            return 0
        return min_sum_length
